app/controllers/page_controller.rb

```diff
@@ -90,41 +101,39 @@ def collect_paginated_components(page_components)
       news_items << pc if pc.component_type === 'PageNewsComponent'
     end
 
-    set_pagy_practice_list_array(practice_lists)
-    paginate_components(events, "events", 3)
-    paginate_components(news_items, "news", 6)
+    set_pagy_practice_list_array(practice_lists) if practice_lists.present?
+    paginate_components(events, "events", 3) if events.present?
+    paginate_components(news_items, "news", 6) if news_items.present?
   end
 
+  # Paginate adjacent news or event components
   def paginate_components(page_components, component_type, pagination)
     return unless page_components
+
     page_item_list_index = 0
-    params_index = params["#{page_item_list_index}"]
+    params_index = params[page_item_list_index.to_s]
     items = []
     ids = []
+    position = page_components.first&.position
+    ids << page_components.first.component_id
```

Comment:
Nit: Should we have a safe navigator on this line as well? Similar to line `117`?

Reply:
Probably can't hurt! This feature wouldn't work if the components didn't have an ID, but it would at least fail quietly with the safe navigator. I'll throw another commit on here.